src/PKSim.Core/Services/ParameterUpdater.cs

```diff
@@ -73,18 +74,40 @@ public ICommand UpdateValue(IParameter sourceParameter, IParameter targetParamet
       private ICommand setTableFormulaValue(IParameter sourceParameter, IParameter targetParameter)
       {
          //target formula is alwyays a table:
+         var targetTableFormula = targetParameter.Formula as DistributedTableFormula;
 
          //updating a table formula from a distributed parameter? only for distributedTableFormula
-         var targetTableFormula = targetParameter.Formula as DistributedTableFormula;
          var sourceDistributedParameter = sourceParameter as IDistributedParameter;
+
          if (targetTableFormula != null && sourceDistributedParameter != null)
-            return _parameterTask.UpdateDistributedTableFormula(targetParameter, sourceDistributedParameter);
+            return updateDistributedTableFormula(targetParameter, sourceDistributedParameter);
 
          //if source parameter as a constant formula, nothing to update as it is a simple growth table
          if (!sourceParameter.Formula.IsTable())
             return null;
 
-         return _parameterTask.UpdateTableFormula(targetParameter, sourceParameter.Formula.DowncastTo<TableFormula>());
+         return updateTableFormula(sourceParameter, targetParameter);
+      }
+
+      private ICommand updateTableFormula(IParameter sourceParameter, IParameter targetParameter)
+      {
+         return withUpdatedValueOrigin(_parameterTask.UpdateTableFormula(targetParameter, sourceParameter.Formula.DowncastTo<TableFormula>()), sourceParameter, targetParameter);
+      }
+
+      private ICommand updateDistributedTableFormula(IParameter targetParameter, IDistributedParameter sourceDistributedParameter)
+      {
+         return withUpdatedValueOrigin(_parameterTask.UpdateDistributedTableFormula(targetParameter, sourceDistributedParameter), sourceDistributedParameter, targetParameter);
+      }
+
+      private ICommand withUpdatedValueOrigin(ICommand command, IParameter sourceParameter, IParameter targetParameter)
+      {
+         if (sourceParameter.ValueOrigin.IsIdenticalTo(targetParameter.ValueOrigin))
```

Comment:
I am only creating the command if the values are indeed different. I don't want to have the history explode with unnecessary commands. Alternatively, this could be done in the SetParameterVAlueOriginTask (but this would go against what we do otherwise)